sdk-setup/src/sdk-assistant

```diff
@@ -192,7 +193,7 @@ while [[ ${1:-} ]]; do
             usage quit
             ;;
         -* )
-            fail "unknown option $1"
+            fail "unknown option $1. Use --help to display usage."
```

Comment:
What about just dumping the help after "unknown option $1"?

Reply:
> What about just dumping the help after "unknown option $1"?

I thought about that as well and tried it first. The help is quite long, so the reason for the failure ("unknown option $1") would not be visible without scrolling up. That's why I chose option b, i.e. hint how to get help.

Reply:
What if help is printed out first, then empty line, and only after that "unknown option $1"

Reply:
Good idea. I changed also the "unknown command" case to work in a similar fashion.

Reply:
As noted in bz it is uncommon to mess up whole screen on bad usage and I hate the few programs that behave so.. Everyone makes a typo sometimes.

I would prefer to follow the well behaving ones:

```
$ cp --foo
cp: unrecognized option '--foo'
Try `cp --help' for more information.
$ zypper --foo
Unknown option '--foo'
$ gcc --foo
bash: gcc: command not found
$ sb2 gcc --foo
i486-meego-linux-gnu-gcc: error: unrecognized command line option ‘--foo’
i486-meego-linux-gnu-gcc: fatal error: no input files
compilation terminated.
$ osc --foo
osc: no such option: --foo
Try 'osc help' for info.
```

Reply:
```
$ sdk-assistant
FAIL: command expected
```
With this kind of tool that doesn't work without a command I think that dumping help for instance in above case would be perfectly good. I understand your concern and I'd not like either if cp/mv would dump some help always when doing a typo but as sdk-assistant is a utility tool. Hence, I'd rather compare it to dnf, zypper and git kind of programs. Further, sdk-assistant is not daily tool for developer per se. Even updating target is not necessary to do that often for most of the developer => people forget these. Hence, I think dumping help is a good idea for it.

Reply:
yup, I can live with it dumping help when no argument is given :)

Reply:
It will now print the --help hint always with fail

Reply:
@vigejolla Raine wished to get the full help dumped when no argument is given. Would you add that?

Reply:
> @vigejolla Raine wished to get the full help dumped when no argument is given. Would you add that?

sure (I'll do it later today)

Reply:
done